### PR TITLE
Add Symbol.isSuperAccessor to reflection API

### PR DESCRIPTION
**Pull request by pikinier20**

### CODE REVIEW

File: scaladoc/src/dotty/tools/scaladoc/tasty/ClassLikeSupport.scala [hunk 1]
```diff
@@ -249,7 +249,7 @@ trait ClassLikeSupport:
   extension (c: ClassDef)
     def extractMembers: Seq[Member] = {
       val inherited = c.getNonTrivialInheritedMemberTrees.collect {
-        case dd: DefDef if !dd.symbol.isClassConstructor && !(dd.symbol.isSuperBridgeMethod || dd.symbol.isDefaultHelperMethod) => dd
+        case dd: DefDef if !dd.symbol.isClassConstructor && !(dd.symbol.isSuperAccessor || dd.symbol.isDefaultHelperMethod) => dd
```
Comment (smarter) [inline scaladoc/src/dotty/tools/scaladoc/tasty/ClassLikeSupport.scala:252]: Alternatively, we could make sure to set the Artifact flag (which doesn't do anything at runtime but is used to signal to javac that this method should not be called from user code) on super accesors, default getters, etc.

**Comment by nicolasstucki** [comment]
Needs tests

**Comment by nicolasstucki** [comment]
Part of #13371

**Comment by nicolasstucki** [comment]
Rebased on master

**Review by nicolasstucki — COMMENTED** [review]
@pikinier20 could you try the alternative solution that does not require this addition?

**Comment by pikinier20** [comment]
@nicolasstucki Do you think it's worth to adding such method to API?

**Comment by nicolasstucki** [comment]
I think it is worth it even if there is another if later we manage to also do this via flags. We will still have code generated before the addition of the flag that will need to fall back to this check.

@pikinier20 you can remove the MiMa filters as they are not needed anymore. Instead you have to run locally `scala3-bootstrapped/testCompilation tests/run-custom-args/tasty-inspector/stdlibExperimentalDefinitions.scala` and update the list there.

**Comment by ckipp01** [comment]
What's the current status with this. Are you still waiting on a review @pikinier20? @nicolasstucki is this something you can return back to to review?

**Comment by nicolasstucki** [comment]
I think we should rebase this and merge it. It looks good to me.

**Review by nicolasstucki — APPROVED** [review]

**Pull request merged**